documentation/wiki/MSBuild-overview.md

```diff
@@ -0,0 +1,60 @@
+- [Quick intro](#quick-intro)
+- [Language interpretation parts](#language-interpretation-parts)
+- [MSBuild APIs and components](#msbuild-apis-and-components)
+- [MSBuild execution modes](#msbuild-execution-modes)
+
+# Quick intro
+MSBuild implements an actual language (... MSBuildian?). For syntax it uses XML. The XML elements and attributes represent keywords, variable names, and expressions in the language. The language is interpreted.
+
+MSBuild has two data structures:
+- [properties](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-properties): single valued, scalars. Like `string foo` in C#
+- [items](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-items): multi valued, vectors. Like `string[] foo` in C#.
```

Comment:
Not really like a string arrays because they can have metadata

Reply:
True, but my intent was to keep this is as a big overview that gives one the "feel of the landscape", not an exhaustive specification. I'll describe this intent in the introduction. There's also the hyperlink to the VS docs with all the details.

Reply:
IMHO this is a core concept -- if you don't want to mention it here I'd avoid analogizing with `string[]` as it's misleading.

Reply:
The main aspect I wanted to convey was that: "there's two types, one that represents a single thing, and another that's a list of things". Maybe `object foo` and `object[] foo`? :) Or maybe `ItemType[] foo` where `ItemType` is `{string ItemValue, Dict<string, string> Metadata}`

Reply:
```suggestion
- [items](https://docs.microsoft.com/en-us/visualstudio/msbuild/msbuild-items): multi valued, vectors. A little `string[] foo` in C#, except the array has a name called an item-type, and each item may not only have a value, but also have associated key-value pairs known as metadata.
```
?